Nodejs/Product/Nodejs/TypingsAcquisition.cs

```diff
@@ -0,0 +1,123 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+
+using SR = Microsoft.NodejsTools.Project.SR;
+
+namespace Microsoft.NodejsTools {
+    static class TypingsAcquisition {
+        private const string TsdExe = "tsd.cmd";
+        private const string TypingsDirecctoryName = "typings";
+
+        public static async Task<bool> AcquireTypings(
+            string pathToRootNpmDirectory,
+            string pathToRootProjectDirectory,
+            IEnumerable<IPackage> packages,
+            Redirector redirector) {
+            var newPackages = TypingsToAcquire(pathToRootProjectDirectory, packages);
+            return await DownloadTypings(
+                pathToRootNpmDirectory,
+                pathToRootProjectDirectory,
+                newPackages,
+                redirector);
+        }
+
+        private static async Task<bool> DownloadTypings(
+            string pathToRootNpmDirectory,
+            string pathToRootProjectDirectory,
+            IEnumerable<IPackage> packages,
+            Redirector redirector) {
+            if (!packages.Any()) {
+                return true;
+            }
+
+            var tsdPath = GetTsdPath(pathToRootNpmDirectory);
+            if (tsdPath == null) {
+                if (redirector != null)
+                    redirector.WriteErrorLine(SR.GetString(SR.TsdNotInstalledError));
+                return false;
+            }
+
+            using (var process = ProcessOutput.Run(
+                tsdPath,
+                TsdInstallArguments(packages),
+                pathToRootProjectDirectory,
+                null,
+                false,
+                redirector,
+                quoteArgs: true)) {
+                if (!process.IsStarted) {
+                    // Process failed to start, and any exception message has
+                    // already been sent through the redirector
+                    if (redirector != null) {
+                        redirector.WriteErrorLine("could not start tsd");
+                    }
+                    return false;
+                }
+                var i = await process;
+                if (i == 0) {
+                    if (redirector != null) {
+                        redirector.WriteLine(SR.GetString(SR.TsdInstallCompleted));
+                    }
+                    return true;
+                } else {
+                    process.Kill();
+                    if (redirector != null) {
+                        redirector.WriteErrorLine(SR.GetString(SR.TsdInstallErrorOccurred));
+                    }
+                    return false;
+                }
+            }
+        }
+
+        private static string GetTsdPath(string pathToRootNpmDirectory) {
+            var path = Path.Combine(pathToRootNpmDirectory, TsdExe);
+            return File.Exists(path) ? path : null;
+        }
+
+        private static string GetPackageTsdName(IPackage package) {
+            return package.Name;
+        }
```

Comment:
nit: Can you include these below the method that is invoking them? Makes it easier to read through the code.